### PR TITLE
feat: group_by sales partner in accounts receivable

### PR DESCRIPTION
**Pull request by rvpasari**
https://app.asana.com/0/1192407975826276/1199549207931835

### CODE REVIEW

**Comment by rvpasari** [comment]
AR Sales Partner

File: erpnext/accounts/report/accounts_receivable/accounts_receivable.py [hunk 1]
```diff
@@ -88,6 +89,59 @@ def get_data(self):
...
+					"paid": paid_amount,
+					"invoiced": invoiced_amount,
+					"range1": range1,
+					"range2": range2,
+					"range3": range3,
+					"range4": range4,
+					"total": total
+				}
+				data.append(summary_line)
+				data.append({})
```
Comment (hrwX) [inline erpnext/accounts/report/accounts_receivable/accounts_receivable.py:126]:
```suggestion
```

File: erpnext/accounts/report/accounts_receivable/accounts_receivable.py [hunk 2]
```diff
@@ -255,6 +310,7 @@ def set_invoice_details(self, row):
 		invoice_details = self.invoice_details.get(row.voucher_no, {})
 		if row.due_date:
 			invoice_details.pop("due_date", None)
+		row['sales_partner'] = invoice_details.get("sales_partner","")
```
Comment (hrwX) [inline erpnext/accounts/report/accounts_receivable/accounts_receivable.py]:
```suggestion
		row['sales_partner'] = invoice_details.get("sales_partner", "")
```

File: erpnext/accounts/report/accounts_receivable/accounts_receivable.py [hunk 3]
```diff
@@ -620,6 +676,11 @@ def prepare_conditions(self):
 		self.add_accounting_dimensions_filters(conditions, values)
 		return " and ".join(conditions), values
 
+	def order_by_partner(self):
+		if self.filters.get("order_by_partner"):
+			return "order by sales_partner, posting_date"
```
Comment (hrwX) [inline erpnext/accounts/report/accounts_receivable/accounts_receivable.py]: - where is this function used ?

**Review by hrwX — DISMISSED** [review]

File: erpnext/accounts/report/accounts_receivable/accounts_receivable.py (hunk 1 above)
Comment (rvpasari) [reply erpnext/accounts/report/accounts_receivable/accounts_receivable.py:126]: We need an empty line after the summary line, hence I added this.

File: erpnext/accounts/report/accounts_receivable/accounts_receivable.py (hunk 3 above)
Comment (rvpasari) [reply erpnext/accounts/report/accounts_receivable/accounts_receivable.py]: not used, deleting

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**